pkg/apis/application/v1alpha1/types_test.go

```diff
@@ -405,7 +405,136 @@ func TestAppProjectSpec_DestinationClusters(t *testing.T) {
 	}
 }
 
-func TestRepository_CopyCredentialsFrom(t *testing.T) {
+func TestRepository_HasCredentials(t *testing.T) {
+
+	tests := []struct {
+		name string
+		repo Repository
+		want bool
+	}{
+		{
+			name: "TestHasRepo",
+			repo: Repository{Repo: "foo"},
+			want: false,
+		},
+		{
+			name: "TestHasUsername",
+			repo: Repository{Username: "foo"},
+			want: true,
+		},
+		{
+			name: "TestHasPassword",
+			repo: Repository{Password: "foo"},
+			want: true,
+		},
+		{
+			name: "TestHasSSHPrivateKey",
+			repo: Repository{SSHPrivateKey: "foo"},
+			want: true,
+		},
+		{
+			name: "TestHasTLSClientCertData",
+			repo: Repository{TLSClientCertData: "foo"},
+			want: true,
+		},
+		{
+			name: "TestHasInsecureHostKey",
+			repo: Repository{InsecureIgnoreHostKey: true},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.repo.HasCredentials(); got != tt.want {
+				t.Errorf("Repository.HasCredentials() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestRepository_IsInsecure(t *testing.T) {
+	tests := []struct {
+		name string
+		repo Repository
+		want bool
+	}{
+		{
+			name: "TestHasRepo",
+			repo: Repository{Repo: "foo"},
+			want: false,
+		},
+		{
+			name: "TestHasUsername",
+			repo: Repository{Username: "foo"},
+			want: false,
+		},
+		{
+			name: "TestHasInsecure",
+			repo: Repository{Insecure: true},
+			want: true,
+		},
+		{
+			name: "TestHasInsecureHostKey",
+			repo: Repository{InsecureIgnoreHostKey: true},
+			want: true,
+		},
+		{
+			name: "TestHasEnableLFS",
+			repo: Repository{EnableLFS: true},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.repo.IsInsecure(); got != tt.want {
+				t.Errorf("Repository.IsInsecure() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestRepository_IsLFSEnabled(t *testing.T) {
+	tests := []struct {
+		name string
+		repo Repository
+		want bool
+	}{
+		{
+			name: "TestHasRepo",
+			repo: Repository{Repo: "foo"},
+			want: false,
+		},
+		{
+			name: "TestHasUsername",
+			repo: Repository{Username: "foo"},
+			want: false,
+		},
+		{
+			name: "TestHasInsecure",
+			repo: Repository{Insecure: true},
+			want: false,
+		},
+		{
+			name: "TestHasInsecureHostKey",
+			repo: Repository{InsecureIgnoreHostKey: true},
+			want: false,
+		},
+		{
+			name: "TestHasEnableLFS",
+			repo: Repository{EnableLFS: true},
+			want: true,
+		},
+	}
+	for _, tt := range tests {
```

Comment:
I think you could simply this test to two lines, as any field other than `EnableLFS`should not matter